crates/oxc_parser/src/js/statement.rs

```diff
@@ -27,19 +27,22 @@ impl<'a> Parser<'a> {
 
         let mut expecting_diretives = true;
         while !self.at(Kind::Eof) {
+            let start_span = self.start_span();
+
             match self.cur_kind() {
                 Kind::RCurly if !is_top_level => break,
                 Kind::Import if !matches!(self.peek_kind(), Kind::Dot | Kind::LParen) => {
                     let stmt = self.parse_import_declaration()?;
                     statements.push(stmt);
                 }
                 Kind::Export => {
-                    let stmt = self.parse_export_declaration()?;
+                    let stmt = self.parse_export_declaration(start_span)?;
                     statements.push(stmt);
                 }
-                Kind::At => {
+                Kind::At if self.at_decorator_before_export() => {
                     self.eat_decorators()?;
-                    continue;
+                    let stmt = self.parse_export_declaration(start_span)?;
+                    statements.push(stmt);
                 }
```

Comment:
This isn't the most efficient way. It basically looks ahead all the decorators to check if it is the "decorator before export" scenario and could result in the decorators being parsed twice in cases like `@dec1 class A {}`.

I made the explicit distinction here to avoid accepting a `start_span` argument in the `parse_statement_list_item` function, which handles decorators in itself.